src/lib/EventPublisher.js

```diff
@@ -18,119 +18,84 @@
 *
 */
 
+import type { Event, EventData } from '../types';
 
 import EventEmitter from 'events';
+import moment from 'moment';
 import logger from './logger';
 import nullthrows from 'nullthrows';
+import uuid from './uuid';
 
-class EventPublisher extends EventEmitter {
-  _eventHandlerByKey: Map<string, Function> = new Map();
-
-  getEventKey(name: ?string, userId: string, coreId: ?string): string {
-    let eventKey = userId;
-    if (coreId) {
-        eventKey += '_' + coreId;
-    }
+const getEventName = (name: ?string, deviceID: ?string): string => {
+  let eventName = '';
+  if (deviceID) {
+    eventName = deviceID;
     if (name) {
-        eventKey += '_' + name;
+      eventName += '/' + name;
     }
+  } else if (name){
+    eventName = name;
+  }
 
-    return eventKey;
+  if (!eventName) {
+    return "*all*";
   }
 
-  getEventName(name: ?string, coreId: ?string){
-    let eventName = '';
-    if (coreId) {
-        eventName = coreId;
-        if (name) {
-          eventName += '/' + name;
-        }
-    } else if (name){
-        eventName = name;
-    }
+  return eventName;
+};
 
-    if (!eventName) {
-        return "*all*";
-    }
+type Subscription = {
+  eventName: string,
+  eventHandler: (event: Event) => void,
+}
 
-    return eventName;
-  }
+class EventPublisher extends EventEmitter {
+  _subscriptionsByID: Map<string, Subscription> = new Map();
 
   publish = (
-    isPublic: boolean,
-    name: string,
-    userId: ?string,
-    data: string,
-    ttl: number,
-    publishedAt: Date,
-    coreId: string,
-  ): void => {
-    const params = [isPublic, name, userId, data, ttl, publishedAt, coreId];
+    eventData: EventData,
+  ): Promise<void> => new Promise((resolve: () => void) => {
```

Comment:
Why are you returning a promise here? Even if the implementation was changed, nothing should care if this function is synchronous or asynchronous since it returns `void`.